src/machine/machinectl.c

```diff
@@ -2826,6 +2918,18 @@ static int parse_argv(int argc, char *argv[]) {
                                 return log_oom();
                         break;
 
+                case ARG_NUMBER_IPS:
+                        if (streq(optarg, "all"))
+                                arg_addrs = ALL_IP_ADDRESSES;
+                        else if (safe_atoi32(optarg, &arg_addrs) < 0) {
```

Comment:
`arg_addrs` is of type "int", not "int32_t", hence please use `safe_atoi()` instead of `safe_atoi32`.